src/Eto/TypeHelper.cs

```diff
@@ -14,9 +14,15 @@ namespace Eto
 	static class TypeHelper
 	{
 		#if PCL
-		static MethodInfo getCallingAssembly = typeof(Assembly).GetTypeInfo().GetDeclaredMethod("GetCallingAssembly");
-
-		public static MethodInfo GetCallingAssembly { get { return getCallingAssembly; } }
+        public static Assembly GetCallingAssembly()
+        {
+            var getCallingAssembly = typeof(Assembly).GetTypeInfo().GetDeclaredMethod("GetCallingAssembly");
+            var detectType = Type.GetType("Eto.PlatformDetect, Eto.Gtk", false);
+            if (detectType != null)
+                getCallingAssembly = detectType.GetRuntimeMethod("GetCallingAssembly", new Type[] { });
+
+            return getCallingAssembly.Invoke(null, null) as Assembly;
```

Comment:
afaik, this will always return Eto.dll, since it's being called from Eto code..  hence is why the .Invoke was repeated everywhere before.

Reply:
You have not looked at the 2nd part of code, its not invoking an actual GetCallingAssembly, but instead a StackTrace.

Reply:
Oh, I understand what you meant now, will fix it ASAP.